File: CleanerAddisApi/CleanAddisBackOfficeApi/src/main/java/com/iyoa/cleanaddis/model/Post.java
package com.iyoa.cleanaddis.model;

import java.sql.Date;
import java.util.UUID;

import javax.persistence.*;
import javax.persistence.Entity;
import javax.persistence.Table; 


@Entity
@Table(name = "post")
public class Post {
	enum PostStatus {
		   POSTED, REMOVEDBYREPORT,DELETED,EDITED
	}
	enum CanBeViewedBy {
		    FRIENDS,FRIENDSOFFRIENDS,MEONLY,ALL
	}
		
	@Id
	@Column(name="uuid")
	private UUID uuid;
	
	@Column(name="username")
	private String username;
	
	@Column(name="media_uuid")
	private UUID mediaUuid;
	
	@Column(name="date")
	private Date date;
	
	@Column(name="noLike")
	private int noLike;
	
	@Column(name="no_view")
	private int noView;
	
	@Column(name="status")
	private PostStatus status;
	
	@Column(name="downloadable")
	private int downloadable;
	
	@Column(name="can_be_viewed_by")
	private CanBeViewedBy canBeViewedBy;
	
	@Column(name="allow_to_be_used_for_article")
	private int allow_to_be_used_for_article;	
}
 

File: app/src/main/java/com/iyoa/cleanaddis/entity/posting/Post.kt
package com.iyoa.cleanaddis.entity.posting


import androidx.room.Entity
import androidx.room.PrimaryKey
import java.io.Serializable
import java.util.*


 data class Post ( val username:String,
                 val noLike:Int, val noView: Int,  val downloadable: Int,
                 val allowToBeUsedForArticle:Int,
                 val mediaUuid:String, val categoryUuid:String,val canBeViewedBy:String,
                 val status: String ,
                 val date:String)
